docs/efficient_gas_calculation_algorithm.md

```diff
@@ -0,0 +1,156 @@
+# Efficient gas calculation algorithm for EVM
+
+This article describes how to efficiently calculate gas and check stack requirements
+for Ethereum Virtual Machine (EVM) instructions.
+
+
+## Instructions metadata
+
+Let's start by defining some basic and universal instructions' parameters.
+
+1. Base gas cost.
+   
+   This is the static gas cost of instructions. Some instructions may have 
+   additional cost depending on their operand values - these have to be
```

Comment:
What about those whose gas costs also depend on the state?

Reply:
The same, you only include the minimal possible cost here. E.g. for `SSTORE` the base cost is 0 (old HFs) or 5000. And later might go to 200 with EIP-2200.

Reply:
Can we assume the base gas cost to never be higher than the final gas cost?

Reply:
yes

Reply:
Can you add that to the text?

Reply:
> Can we assume the base gas cost to never be higher than the final gas cost?

In some way this is the definition of it.

Reply:
I'm just saying that 'additional cost' could be interpreted as 'cost that has to be added' in which case you could also add a negative value. Furthermore, the sentence only explicitly states "depending on their operand values". I think it would be clearer if you also state "on their operand values or the environment", in which case there is no doubt left. Just my two cents :)

Reply:
Added both suggestions in the best way I was capable of.